runtime/common/src/purchase.rs

```diff
@@ -0,0 +1,989 @@
+
+			// Verify the signature provided is valid for the statement.
+			Self::verify_signature(&who, &signature)?;
+
+			// Create a new pending account.
+			let status = AccountStatus {
+				validity: AccountValidity::Initiated,
+				signature,
+				free_balance: Zero::zero(),
+				locked_balance: Zero::zero(),
+				vat: Permill::zero(),
+			};
+			Accounts::<T>::insert(&who, status);
+			Self::deposit_event(RawEvent::AccountCreated(who));
+		}
+
+		/// Update the validity status of an existing account. If set to completed, the account
+		/// will no longer be able to continue through the crowdfund process.
+		///
+		/// We check tht the account exists at this stage, but has not completed the process.
+		///
+		/// Origin must match the `ValidityOrigin`.
+		#[weight = T::DbWeight::get().reads_writes(1, 1)]
+		fn update_validity_status(origin,
+			who: T::AccountId,
+			validity: AccountValidity
+		) {
+			T::ValidityOrigin::ensure_origin(origin)?;
+			ensure!(Accounts::<T>::contains_key(&who), Error::<T>::InvalidAccount);
+			Accounts::<T>::try_mutate(&who, |status: &mut AccountStatus<BalanceOf<T>>| -> DispatchResult {
+				ensure!(status.validity != AccountValidity::Completed, Error::<T>::AlreadyCompleted);
+				status.validity = validity;
+				Ok(())
+			})?;
+			Self::deposit_event(RawEvent::ValidityUpdated(who, validity));
+		}
+
+		/// Update the balance of a valid account.
+		///
+		/// We check tht the account is valid for a balance transfer at this point.
+		///
+		/// Origin must match the `ValidityOrigin`.
+		#[weight = T::DbWeight::get().reads_writes(1, 1)]
+		fn update_balance(origin,
+			who: T::AccountId,
+			free_balance: BalanceOf<T>,
+			locked_balance: BalanceOf<T>,
+			vat: Permill,
+		) {
+			T::ValidityOrigin::ensure_origin(origin)?;
+
+			Accounts::<T>::try_mutate(&who, |status: &mut AccountStatus<BalanceOf<T>>| -> DispatchResult {
+				// Account has a valid status (not Invalid, Pending, or Completed)...
+				ensure!(status.validity.is_valid(), Error::<T>::InvalidAccount);
+				free_balance.checked_add(&locked_balance).ok_or(Error::<T>::Overflow)?;
+				status.free_balance = free_balance;
+				status.locked_balance = locked_balance;
+				status.vat = vat;
+				Ok(())
+			})?;
+			Self::deposit_event(RawEvent::BalanceUpdated(who, free_balance, locked_balance));
+		}
+
+		/// Pay the user and complete the purchase process.
+		///
+		/// We reverify all assumptions about the state of an account, and complete the process.
+		///
+		/// Origin must match the configured `PaymentAccount`.
+		#[weight = T::DbWeight::get().reads_writes(4, 2)]
+		fn payout(origin, who: T::AccountId) {
+			// Payments must be made directly by the `PaymentAccount`.
+			let payment_account = ensure_signed(origin)?;
+			ensure!(payment_account == PaymentAccount::<T>::get(), DispatchError::BadOrigin);
+
+			// Account should not be active.
+			ensure!(system::Module::<T>::is_dead_account(&who), Error::<T>::ExistingAccount);
```

Comment:
Why?

Reply:
I have removed this restriction. Now any existing account without a vesting schedule can participate.